### PR TITLE
YOLOE: Use passed `imgsz` for visual prompt extraction

### LINKED ISSUE #22233: YOLOE Visual Prompt ROI Discrepancy
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

When re-detecting on the reference image itself. Why Detected Bounding Boxes Differ from Prompted ROI and the confidence is very low?

### Additional

When using YOLOE with visual prompts, I observed that the detected object's bounding box significantly differs from and encompasses a larger area than the ROI specified in visual_prompts, even when re-detecting on the reference image itself. This behavior is  noticeable for a circular object where the manually defined ROI intended to tightly enclose it.
For instance, I provided the following visual_prompts:
`
import numpy as np
import torch
import torch.nn.functional as F  
from ultralytics import YOLOE
from ultralytics.models.yolo.yoloe import YOLOEVPSegPredictor,YOLOEVPDetectPredictor

model = YOLOE("yoloe-v8s-seg.pt")
visual_prompts = dict(
    bboxes=np.array(
        [
            [37.21,40.64,101.2,102.7],  
        ],
    ),
    cls=np.array(
        [
            0,  
        ]
    ),
)

results = model.predict(
    "boss.jpg",  
    refer_image="boss.jpg",  
    visual_prompts=visual_prompts,
    predictor=YOLOEVPSegPredictor,
    #save=True,
    exist_ok=True,
)
results[0].show(labels=False)
`
the resulting bounding box for the detected object often includes more surrounding area than the ROI, as illustrated in the image below. and the confidence is very low

Image

Image

### PR DESCRIPTION
**Pull request by Y-T-G**
Fixes #22233 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Adds imgsz propagation to model export during predict for consistent image sizing, and refreshes YOLO26 docs with clearer wording and formatting. 🚀

### 📊 Key Changes
- Code: Passes imgsz into export args in `YOLO.predict()` so exported models use the requested or overridden image size, not a default.
- Docs: Polishes YOLO26 documentation (consistent apostrophes, spacing, and phrasing), keeping all previews and metrics the same.

### 🎯 Purpose & Impact
- More reliable exports: Ensures ONNX/TensorRT (and other exports) match your specified `imgsz`, preventing silent size mismatches that can affect accuracy, speed, and memory. 📐
- Better UX: Reduces confusion when switching between training/inference sizes and exporting, improving reproducibility. ✅
- Clearer docs: Easier-to-read YOLO26 page while maintaining current preview status and guidance. 📖

Helpful link:
- Read the updated YOLO26 page: YOLO26 model overview and previews (https://docs.ultralytics.com/models/yolo26/)

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response — an Ultralytics engineer will review and assist you shortly. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

Since this PR appears to address a bug and references issue #22233, please ensure a Minimum Reproducible Example (MRE) is included or linked in your PR description. For example:
- Python snippet showing current behavior vs. expected behavior with `imgsz` during `model.predict()`, e.g.:
```python
from ultralytics import YOLO
model = YOLO("yolo11n.pt")
model.predict("image.jpg", imgsz=896)  # demonstrate the issue and the fix behavior
```
- CLI reproduction steps if applicable
- Any logs or console output that illustrate the problem and resolution

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🎯✨

**Review by Laughing-q — APPROVED** [review]

**Comment by UltralyticsAssistant** [comment]
Merged with excitement — thank you, @Y-T-G and @Laughing-q! 

“Details make perfection, and perfection is not a detail.” — Leonardo da Vinci. Your imgsz propagation in `YOLO.predict()` brings that precision to exports, ensuring ONNX/TensorRT models match user intent for size, performance, and reproducibility. The refreshed YOLO26 docs are a clear step up in readability, too.

Appreciate the craftsmanship and teamwork here. Readers can explore the updated docs at the YOLO26 model overview and previews: https://docs.ultralytics.com/models/yolo26/ 🚀
